test/e2e/testsuites/failed_mount.go

```diff
@@ -152,13 +174,26 @@ func (t *gcsFuseCSIFailedMountTestSuite) DefineTests(driver storageframework.Tes
 		ginkgo.By("Deleting the Kubernetes service account")
 		testK8sSA.Cleanup(ctx)
 
+		// For invalid SA testcase, The Unauthenticated error spawns from prepareStorageClient() in CSI NodePublish. When CSI skips bucket access check, this step is skipped.
+		if configPrefix == specs.SkipCSIBucketAccessCheckPrefix {
+			return
+		}
+
 		ginkgo.By("Checking that the pod has failed mount error Unauthenticated")
 		tPod.WaitForFailedMountError(ctx, codes.Unauthenticated.String())
 		tPod.WaitForFailedMountError(ctx, "storage service manager failed to setup service: context deadline exceeded")
+	}
+
+	ginkgo.It("should fail when the specified service account does not have access to the GCS bucket", func() {
+		testCaseSAInsufficientAccess("")
 	})
 
-	ginkgo.It("should fail when the sidecar container is not injected", func() {
-		init()
+	ginkgo.It("[csi-skip-bucket-access-check] should fail when the specified service account does not have access to the GCS bucket", func() {
```

Comment:
do you want to add a test case where it starts without access, and then access gets added?

Reply:
Good point! The tricky part is how to give the SA access after it's created. We can reuse the following code: https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/blob/6c96f53e2567fb3ea9a3fa9372aea03ef306b62b/test/e2e/testdriver.go#L354-L360

Reply:
Created an issue to track the test case and unblock this PR
https://github.com/GoogleCloudPlatform/gcs-fuse-csi-driver/issues/271